include/bluetooth/hci.h

```diff
@@ -2442,8 +2442,8 @@ struct bt_hci_evt_le_chan_sel_algo {
 
 #define BT_HCI_EVT_LE_CONNECTIONLESS_IQ_REPORT  0x15
 struct bt_hci_le_iq_sample {
-	int8_t i;
-	int8_t q;
+	int16_t i;
+	int16_t q;
```

Comment:
I and Q samples are signed integer of 1 octet. Refer to BLUETOOTH CORE SPECIFICATION Version 5.3 | Vol 4, Part E Section 7.7.65.21

```suggestion
	int8_t i;
	int8_t q;
```

Reply:
Thank you @cvinayak, yes I checked it now and yes, you are right.
But now there is a problem that all the time, 4 least significant bits are removed. It means if the transmitter is far away from the receiver, probably the received data could be only 4 bits which is shifted and omitted, so no data at all!
The shift operation has executed in this line:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/controller/hci/hci.c#L2681
```
iq_tmp = IQ_SHIFT_12_TO_8_BIT(iq_report->sample[idx].i);
sep->sample[idx].i = (int8_t)iq_tmp;
iq_tmp = IQ_SHIFT_12_TO_8_BIT(iq_report->sample[idx].q);
sep->sample[idx].q = (int8_t)iq_tmp;
```

Also, in practice, the values are not too much and mostly between -300 to 300 (as I have measured by nRF52833-DK boards). So 4 bit shift means omitting an important part of the data while the most significant bits are 0 and doesn't have any important information.

One solution could be selecting 8 bits of the most significant bits which are available (non zero excluding the sign bit). In this scenario, the IQ samples are considered as proportional and for example both 0x015E (350) and 0x0057 (87) will be reported as 0x0057 (87). 
 Another solution could be shifting for example 2 bits and considering the -512 to 511 as the minimum/maximum possible values (higher values will be considered as saturation).